vm/stardust/vm_stardust_test.go

```diff
@@ -1264,6 +1264,1512 @@ func TestStardustTransactionExecution(t *testing.T) {
+							Amount: defaultAmount,
+							Conditions: iotago.BasicOutputUnlockConditions{
+								&iotago.AddressUnlockCondition{Address: ed25519Addresses[1]},
+							},
+						},
+					}
+				},
+				outputsFunc: func(ed25519Addresses []iotago.Address, testAddresses []iotago.Address, inputIDs iotago.OutputIDs, totalInputAmount iotago.BaseToken) iotago.TxEssenceOutputs {
+					return iotago.TxEssenceOutputs{
+						&iotago.BasicOutput{
+							Amount: totalInputAmount,
+							Conditions: iotago.BasicOutputUnlockConditions{
+								&iotago.AddressUnlockCondition{Address: &iotago.MultiAddress{
+									Addresses: []*iotago.AddressWithWeight{
+										{
+											Address: ed25519Addresses[0],
+											Weight:  1,
+										},
+										{
+											Address: ed25519Addresses[1],
+											Weight:  1,
+										},
+									},
+									Threshold: 3,
+								}},
+							},
+						},
+					}
+				},
+				unlocksFunc: func(sigs []iotago.Signature, testAddresses []iotago.Address) iotago.Unlocks {
+					return iotago.Unlocks{
+						&iotago.SignatureUnlock{Signature: sigs[0]},
+						&iotago.SignatureUnlock{Signature: sigs[1]},
+					}
+				},
+				wantEncodeErr:  iotago.ErrMultiAddressThresholdInvalid,
+				wantExecuteErr: nil,
+			}
+		}(),
+
+		// fail - threshold < 1 (needs to be checked on output side)
+		func() test {
+			return test{
+				name:           "fail - threshold < 1 (needs to be checked on output side)",
+				ed25519AddrCnt: 1,
+				addressesFunc:  nil,
+				inputsFunc: func(ed25519Addresses []iotago.Address, testAddresses []iotago.Address) []iotago.Output {
+					return []iotago.Output{
+						&iotago.BasicOutput{
+							Amount: defaultAmount,
+							Conditions: iotago.BasicOutputUnlockConditions{
+								&iotago.AddressUnlockCondition{Address: ed25519Addresses[0]},
+							},
+						},
+					}
+				},
+				outputsFunc: func(ed25519Addresses []iotago.Address, testAddresses []iotago.Address, inputIDs iotago.OutputIDs, totalInputAmount iotago.BaseToken) iotago.TxEssenceOutputs {
+					return iotago.TxEssenceOutputs{
+						&iotago.BasicOutput{
+							Amount: totalInputAmount,
+							Conditions: iotago.BasicOutputUnlockConditions{
+								&iotago.AddressUnlockCondition{Address: &iotago.MultiAddress{
+									Addresses: []*iotago.AddressWithWeight{
+										{
+											Address: ed25519Addresses[0],
+											Weight:  1,
+										},
+									},
+									Threshold: 0,
+								}},
+							},
+						},
+					}
+				},
+				unlocksFunc: func(sigs []iotago.Signature, testAddresses []iotago.Address) iotago.Unlocks {
+					return iotago.Unlocks{
+						&iotago.SignatureUnlock{Signature: sigs[0]},
+					}
+				},
+				wantEncodeErr:  iotago.ErrMultiAddressThresholdInvalid,
+				wantExecuteErr: nil,
+			}
+		}(),
+
+		// fail - address weight == 0 (needs to be checked on output side)
+		func() test {
+			return test{
+				name:           "fail - address weight == 0 (needs to be checked on output side)",
```

Comment:
Nit: This case should have its own error type imo, because it's not really related to the threshold. I.e. don't use `ErrMultiAddressThresholdInvalid` but use a new type `ErrMultiAddressWeightInvalid` or something like that. That would be nicer for debugging, too.